api4/user.go

```diff
@@ -1129,6 +1140,93 @@ func updateUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(ruser.ToJson()))
 }
 
+func patchUserCheckLoginProviderAttributes(c *Context, ouser *model.User, patch *model.UserPatch) {
+	const (
+		firstNameField = "first_name"
+		lastNameField  = "last_name"
+		nicknameField  = "nickname"
+		positionField  = "position"
+		emailField     = "email"
+		usernameField  = "username"
+	)
+	// If LDAP or SAML is used, and any of the following attributes have been specified
+	// in the settings, then their corresponding fields may not be changed
+	LS := &c.App.Config().LdapSettings
+	SS := &c.App.Config().SamlSettings
+	for _, provider := range []struct {
+		AuthService        string
+		FirstNameAttribute *string
+		LastNameAttribute  *string
+		NicknameAttribute  *string
+		PositionAttribute  *string
+		EmailAttribute     *string
+	}{
+		{model.USER_AUTH_SERVICE_LDAP, LS.FirstNameAttribute, LS.LastNameAttribute, LS.NicknameAttribute,
+			LS.PositionAttribute, LS.EmailAttribute},
+		{model.USER_AUTH_SERVICE_SAML, SS.FirstNameAttribute, SS.LastNameAttribute, SS.NicknameAttribute,
+			SS.PositionAttribute, SS.EmailAttribute},
+	} {
+		if ouser.AuthService != provider.AuthService {
+			continue
+		}
+		for _, attr := range []struct {
+			name       string
+			patchValue *string
+			userValue  *string
+			selector   *string
+		}{
+			{firstNameField, patch.FirstName, &ouser.FirstName, provider.FirstNameAttribute},
+			{lastNameField, patch.LastName, &ouser.LastName, provider.LastNameAttribute},
```

Comment:
You know how we had noticed that the front-end disables both first name and last name updates if either of those fields are configured... do you think this logic needs to match that? My only thought for making this logic match that is for consistency with the front-end and with the existing production app. What do you think?

Reply:
Good point. Back-end logic should match front-end logic.